File: apps/ios/BankCore/Networking/APIConfig.swift
import Foundation

enum APIConfig {
    static let host = "bankcore.vortexsoftware.tech"
    static let baseURL = URL(string: "https://bankcore.vortexsoftware.tech")!

    static let lucas = DemoAccount(
        name: "Lucas Mendes",
        taxId: "98765432100",
        password: "teste123456"
    )
    static let maria = DemoAccount(
        name: "Maria Silva",
        taxId: "12345678900",
        password: "teste123456"
    )

    /// Mock UX only — not a ledger field.
    static let mockPixLimitReais: Double = 20_000

    struct DemoAccount {
        let name: String
        let taxId: String
        let password: String
    }

    static func url(_ path: String) -> URL {
        var base = baseURL.absoluteString
        if base.hasSuffix("/") { base.removeLast() }
        let suffix = path.hasPrefix("/") ? path : "/" + path
        guard let url = URL(string: base + suffix) else {
            preconditionFailure("Invalid API path: \(path)")
        }
        return url
    }
}

File: apps/android/app/src/main/java/br/vortex/bankcore/data/ApiConfig.kt
package br.vortex.bankcore.data

object ApiConfig {
    const val HOST = "bankcore.vortexsoftware.tech"
    const val BASE_URL = "https://bankcore.vortexsoftware.tech"
    const val APP_VERSION = "1.10.25"
    const val LEGAL_NAME = "Vortex Software LTDA"
    const val CONTACT_EMAIL = "contato@vortexsoftware.tech"

    data class DemoAccount(
        val name: String,
        val taxId: String,
        val password: String,
        val initials: String,
    )

    val lucas = DemoAccount(
        name = "Lucas Mendes Rocha",
        taxId = "98765432100",
        password = "teste123456",
        initials = "LM",
    )

    val maria = DemoAccount(
        name = "Maria Silva",
        taxId = "12345678900",
        password = "teste123456",
        initials = "MS",
    )
}
